libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx/ptx_helper_functions.h

```diff
@@ -24,12 +25,27 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA_PTX
 
+template <int __n>
+using n32_t = _CUDA_VSTD::integral_constant<int, __n>;
+
+/*************************************************************
+ *
+ * Conversion from generic pointer -> state space "pointer"
+ *
+ **************************************************************/
 inline _LIBCUDACXX_DEVICE _CUDA_VSTD::uint32_t __as_ptr_smem(const void* __ptr)
 {
   // Consider adding debug asserts here.
   return static_cast<_CUDA_VSTD::uint32_t>(__cvta_generic_to_shared(__ptr));
 }
 
+inline _LIBCUDACXX_DEVICE _CUDA_VSTD::uint32_t __as_ptr_dsmem(const void* __ptr)
+{
+  // No difference in implementation to __as_ptr_smem.
+  // Consider adding debug asserts here.
+  return static_cast<_CUDA_VSTD::uint32_t>(__cvta_generic_to_shared(__ptr));
```

Comment:
If we do not intent and implementation differences then we should delegate:
```suggestion
  return __as_ptr_smem(__ptr);
```

Reply:
Good point. This has been changed.